lib/china_regions/helpers/form_helper.rb

```diff
@@ -60,29 +61,44 @@ def region_select(object_name, methods, options = {}, html_options = {})
       private
 
       def get_choices(region_klass, method, preselected_choices, index)
-        choices = (index == 0 ? region_klass.where(nil).collect { |p| [ p.name, p.id ] } : [])
+        return preselected_choices[method] if preselected_choices[method]
+
+        index == 0 ? region_klass.where(nil).collect { |p| [ p.name, p.id ] } : []
```

Comment:
This is hard to understand - go ahead and break this up into an `if` statement. No need to get clever, as fun as the one-line solution is.